client/src/language/languageClient.ts

```diff
@@ -23,13 +23,25 @@ import {
   TransportKind,
   type ServerOptions
 } from 'vscode-languageclient/node'
-
-const getEmbeddedLanguageDocInfos = async (client: LanguageClient, uriString: string, position: Position): Promise<{ uri: string, lineOffset: number }> => {
-  return await client.sendRequest('custom/getEmbeddedLanguageDocInfos', { uriString, position })
+import { RequestMethod, type RequestParams, type RequestResult } from '../lib/src/types/requests'
+import { NotificationMethod, type NotificationParams } from '../lib/src/types/notifications'
+
+const getEmbeddedLanguageDocInfos = async (
+  client: LanguageClient,
+  uriString: string,
+  position: Position
+): RequestResult['EmbeddedLanguageDocInfos'] => {
+  const params: RequestParams['EmbeddedLanguageDocInfos'] = { uriString, position }
```

Comment:
Is it possible to use the dot notation to reference the property? You won't need to change the value here in case you change the value of the property.

Reply:
No, it would not be possible since we are accessing the type of properties. On the opposite, eslint would complain if we do RequestMethod['EmbeddedLanguageDocInfos'] and require to use the dot notation, since it returns a string value instead of a type. I agree in some way it gives an appearance of inconsistency in this case.

That being said, if you make typos, such as `RequestResult['EmbededLanguageDocInfo']` or something, Typescript will tell you the property is invalid. It will also suggest completion items while you are typing. The only thing missing is that I could not think of a way to require the three of them (RequestMethod, RequestParams and RequestResult) to use the same property.